lightning/src/util/test_utils.rs

```diff
@@ -311,41 +311,69 @@ impl<'a> Drop for TestRouter<'a> {
 	}
 }
 
-pub struct TestMessageRouter<'a> {
-	inner: DefaultMessageRouter<
-		Arc<NetworkGraph<&'a TestLogger>>,
-		&'a TestLogger,
-		&'a TestKeysInterface,
-	>,
+pub enum TestMessageRouter<'a> {
```

Comment:
Instead of introducing this enum, can each test that needs to a different `MessageRouter` simply override `NodeCfg::message_router` before calling `create_node_chanmgrs`?

Reply:
Thanks a lot for the suggestion!

I actually experimented with overriding NodeCfg::message_router before going with the enum approach. The main challenge I ran into was that NodeCfg is currently configured with a concrete TestMessageRouter type. Switching it to be generic over the MessageRouter trait would require parameterizing NodeCfg itself, which would ripple into a lot of unrelated parts of the test code.

To avoid that complexity, I chose to modify TestMessageRouter into an enum instead. It felt like a cleaner tradeoff that kept the impact localized.

Let me know if this reasoning makes sense to you. Thanks so much for the pointer!

Reply:
Thanks a lot for the suggestion!

I actually experimented with overriding `NodeCfg::message_router` before going with the enum approach. The main challenge I ran into was that `NodeCfg` is currently configured with a concrete `TestMessageRouter` type. Switching it to be generic over the `MessageRouter` trait would require parameterizing `NodeCfg` itself, which would ripple into a lot of unrelated parts of the test code.

To avoid that complexity, I chose to modify `TestMessageRouter` into an enum instead. It felt like a cleaner tradeoff that kept the impact localized.

Let me know if this reasoning makes sense to you. Thanks so much for the pointer!